src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -406,18 +406,13 @@ using EncodeForUriExpression =
       if (!optLanguageTag.has_value() || !optLanguageRange.has_value()) {
         return Id::makeUndefined();
       }
-      // For LANGMATCHES(), the order of the arguments matter.
-      // The language-range "*" will match any non empty string
-      // (Sparql-standard).
-      const std::string& languageTag = optLanguageTag.value();
-      const std::string& languageRange = optLanguageRange.value();
-      if ((languageTag == "" && languageRange == "*") || languageRange == "") {
+      std::string& languageTag = optLanguageTag.value();
+      std::string& languageRange = optLanguageRange.value();
```

Comment:
See my comment in the implementation of `isLanguageMatch`.

It should probably only be 
` return Id::makeFromBool(optLanguageTag.value(), optLaguageRange.value())`, then you also don't need the `opt` as part of the name.